### DISCUSSION TITLE
New PRs blocked after automerge

### DISCUSSION THREAD

**Discussion by xoxys** [renovatebot/renovate#11037] [Request Help]
Hi,

I have noticed on multiple repos now, that after a digest update PR was auto-merged by Renovate, future digest updates were blocked due to `These are blocked by an existing closed PR and will not be recreated unless you click a checkbox below.` But the PR was never closed unmerged. Examples:

https://github.com/thegeeklab/prometheus-pve-sd/issues/3
https://github.com/thegeeklab/prometheus-pve-sd/pull/19

Any idea why this happens?

**Comment by viceice** [comment]
Please check debug logs, You probably have a branch / pr name collision

**Reply by xoxys** [reply]
If an old PR has the same name but was **merged** and not closed, it will still be blocked on future PRs?

**Reply by viceice** [reply]
Check the debug logs, it will show the issue.

**Reply by xoxys** [reply]
I have already checked the logs thats why Im asking here....

```
DEBUG: Found PR #19(branch="renovate/docker-digests")
DEBUG: Found closed PR with current title(branch="renovate/docker-digests")
DEBUG: Retrieved closed PR list with graphql(branch="renovate/docker-digests")
{
  "prNumbers": [
    1,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    2,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    4,
    5,
    6,
    7,
    8,
    9
  ]
}
DEBUG: Returning from graphql closed PR list(branch="renovate/docker-digests")
DEBUG: Closed PR already exists. Skipping branch.(branch="renovate/docker-digests")
{
  "prTitle": "chore(deps): update docker digests"
}
DEBUG: Merged PR is blocking this branch(branch="renovate/docker-digests")
{
  "pr": 19
}
```

But again, is it expected that PRs with the same title were blocked even if the old PR was merged and **not** closed unmerged? I guess not. The logs tell me `Merged PR is blocking this branch` how can a merged PR block future PRs with the same name?

**Reply by viceice** [reply]
@rarkins Maybe we need to ignore merged pr's here?

https://github.com/renovatebot/renovate/blob/ef2a13f4f28f5b2c163991c0c5d4f53b4f45ea9f/lib/workers/branch/check-existing.ts#L22-L31

**Reply by viceice** [reply]
I'm not sure why a merged pr should block new branches 🤔

**Reply by rarkins** [reply]
It was originally to avoid endless loops with automerging. e.g. let's say there's an automerged update, but it's bad, so you roll it back. Then it would be hard to stop Renovate doing it again. However I think *blocking* is not maybe necessary - we could instead disable automerge (if configured) in case we find an identical merged update already.

However, this seems buggy regardless. A PR like "chore(deps): update docker digests" should be immortal and not block in the first place

**Reply by xoxys** [reply]
Should I create an issue or will you do it on your own?

**Reply by rarkins** [reply]
I have raised #11062 and #11063

**Comment by xoxys** [comment]
Just as additional information. On April 15, everything was working as expected and a PR for grouped docker digest updates was flagged as Immortal as usual for custom grouping. On April 28, this stopped working till now :(

![image](https://user-images.githubusercontent.com/3391958/133052683-2e3b5548-3cb6-4795-84db-b2b813ef51e2.png)

**Reply by rarkins** [reply]
The problem seems to be that a PR which should be immortal is not being treated that way. Are you able to reproduce it in a [minimal reproduction repository](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md)?

**Reply by rarkins** [reply]
I searched github and found a counterexample which seems to be correct: https://togithub.com/owncloud-ci/golang/pull/63

**Reply by xoxys** [reply]
Now I'm fully confused.....

This one **does** create an immortal PR:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "docker": {
    "pinDigests": true
  },
  "packageRules": [
    {
      "groupName": "docker digests",
      "matchDatasources": ["docker"],
      "matchUpdateTypes": ["pin", "digest"]
    }
  ]
}
```

And this one **does not** create an immortal commit:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "docker": {
    "pinDigests": true
  },
  "packageRules": [
    {
      "groupName": "docker digests",
      "matchDatasources": ["docker"],
      "matchUpdateTypes": ["pin", "digest"]
    }
  ],
  "droneci": {
    "enabled": false
  }
}
```


How that?? Why does this snippet affect other docker PRs?

```
 "droneci": {
    "enabled": false
  }
```

**Reply by xoxys** [reply]
Yep... Verified again, after adding the droneci -> enabled config back it breaks

![image](https://user-images.githubusercontent.com/3391958/133487397-dbac01ed-90cc-40a1-ae5f-c00743b7118b.png)

**Reply by viceice** [reply]
Can you create a minimal reproduction please

**Reply by xoxys** [reply]
Already done using the configs I posted above... https://github.com/xoxys/test/pull/14

**Reply by xoxys** [reply]
Removing https://github.com/xoxys/test/blob/master/renovate.json#L14-L16 will make the PR immortal again.

**Reply by rarkins** [reply]
What happens when you remove `pin` from `matchUpdateTypes`? `pin` updates are intended to be grouped together and not with digest updates.

**Reply by xoxys** [reply]
Removed, PR was updated; doesnt change anything

**Reply by rarkins** [reply]
So here's the config as I see it right now:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "docker": {
    "pinDigests": true
  },
  "packageRules": [
    {
      "groupName": "docker digests",
      "matchDatasources": ["docker"],
      "matchUpdateTypes": ["digest"]
    }
  ],
  "droneci": {
    "enabled": false
  }
}
```

And can you confirm that specifically it's this "pin digests" PR Ignore settings which is the problematic one?

[![image](https://user-images.githubusercontent.com/6311784/133571641-51e84154-6da7-4302-8318-26bd9b6cbcab.png)](https://github.com/xoxys/test/pull/14)

**Reply by xoxys** [reply]
I can't follow you...

From your reply:
> The problem seems to be that a PR which should be immortal is not being treated that way

So, the issue is that the PR is not immortal, even if it should be. The problematic setting is 

```
"droneci": {
    "enabled": false
  }
```

If I remove this from the renovate.json the PR **will be** immortal as expected. Question is **why**?

**Reply by xoxys** [reply]
But yes I think the "Ignore" setting is the problematic one as this will definitely block further PRs. As you have also mentioned earlier, these PRs should be immortal, and they are if I remove the droneci config option I have mentioned above. But this droneci setting should have no impact to the Digest pinning PR at all. IMO that's a bug.

**Reply by rarkins** [reply]
As this PR is for pinning digests only, can you remove the package rule and it still reproduces? I'm trying to get it to the minimum.

**Reply by xoxys** [reply]
I have removed the entire package rule now. PR is still not immortal and ignored instead.

**Reply by rarkins** [reply]
Here's the bit in our code which is responsible:

drone disabled:
![image](https://user-images.githubusercontent.com/6311784/133581542-041e1c08-a1e8-48cd-a3db-db074460a428.png)

drone enabled:
![image](https://user-images.githubusercontent.com/6311784/133581332-6f53a61e-889f-440b-bcbe-dbb0ded6c00b.png)

But I'm not sure of exactly how to fix it. Can you create a bug report now and link to your repo. The title can be, "Digest pinning PRs should always be immortal"

**Reply by xoxys** [reply]
https://github.com/renovatebot/renovate/issues/11772

**Reply by viceice** [reply]
i know the cause, the drone image is `alpine` so `currentValue` is undefined. That's why we see two `toVersions` here.

@rarkins shouldn't we always set `upgrade.recreateClosed = true` when `updateType=pin`?
